### PR TITLE
Added serviceId params to request

### PR DESCRIPTION
**Pull request by OgunyemiO**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EXUI-812 

### Change description ###

Pass serviceId to LOV API during case linking

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-view/case-view.model.ts [hunk 1]
```diff
@@ -41,3 +41,18 @@ export class CaseView {
   };
   public case_flag?: Flags;
 }
+
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-view/case-view.model.ts:44]: Please put this in separate model file.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts [hunk 2]
```diff
@@ -102,6 +101,13 @@ export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent
     });
   }
 
+  getOrgService(): void {
+    const servicesApiUrl = `refdata/location/orgServices?ccdCaseType=${this.caseDetails.case_type.id}`;
+    this.commonDataService.getServiceOrgData(servicesApiUrl).subscribe(result => {
+      this.getLinkedCaseReasons(result[0].service_id);
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts:107]: Unfortunately you need to handle each of the case types returned. A single case type can contain cases from more than one service. Civil do this, I wish they hadn't but we can't just take the first

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts [hunk 3]
```diff
@@ -39,11 +39,10 @@ export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent
     // Clear validation errors
     this.caseEditDataService.clearFormValidationErrors();
     // Get linked case reasons from ref data
-    this.getLinkedCaseReasons();
     this.linkedCasesService.editMode = false;
-    this.caseEditDataService.caseDetails$.subscribe({
-      next: caseDetails => this.initialiseCaseDetails(caseDetails)
+    this.caseEditDataService.caseDetails$.subscribe({next: caseDetails => { this.initialiseCaseDetails(caseDetails);  }
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts:43]: You need to unsubscribe from all your subscriptions
Kinda ugly on one line like this

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts [hunk 4]
```diff
@@ -89,8 +88,8 @@ export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent
     }
   }
 
-  public getLinkedCaseReasons(): void {
-    const reasonCodeAPIurl = `${this.appConfig.getRDCommonDataApiUrl()}/lov/categories/CaseLinkingReasonCode`;
+  public getLinkedCaseReasons(serviceId: number): void {
+    const reasonCodeAPIurl = `${this.appConfig.getRDCommonDataApiUrl()}/lov/categories/CaseLinkingReasonCode?serviceId=${serviceId}`;
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts:92]: I think you'll have to change the signature to accept an array and call lov/categories for each service ID - though it's worth checking if the parameter accepts a comma separated list (check the source of the lov api in github

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Need to handle multiple service IDs per case type

File: projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-view/case-view.model.ts (hunk 1 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-view/case-view.model.ts]: @andywilkinshmcts . Done

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts (hunk 4 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts:95]: @andywilkinshmcts I require more clarification from you on this. Better to chat over the phone

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts (hunk 2 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts:110]: @andywilkinshmcts I require more clarification from you on this. Better to chat over the phone

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts [hunk 5]
```diff
@@ -102,6 +104,13 @@ export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent
     });
   }
 
+  getOrgService(): void {
+    const servicesApiUrl = `refdata/location/orgServices?ccdCaseType=${this.caseDetails.case_type.id}`;
+    this.commonDataService.getServiceOrgData(servicesApiUrl).subscribe(result => {
+      this.getLinkedCaseReasons(result[0].service_id);
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts:110]: result.foreach(r -> this.getLinkedCaseReasons(r.service_id));

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts [hunk 6]
```diff
@@ -102,6 +104,13 @@ export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent
     });
   }
 
+  getOrgService(): void {
+    const servicesApiUrl = `refdata/location/orgServices?ccdCaseType=${this.caseDetails.case_type.id}`;
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts:108]: add in
 this.linkedCasesService.linkCaseReasons = [];

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by olusegz07 — APPROVED** [review]

**Review by andywilkinshmcts — APPROVED** [review]

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by olusegz07 — APPROVED** [review]

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by olusegz07 — APPROVED** [review]

**Review by richardhenryash — DISMISSED** [review]

**Pull request merged**